docs/pages/router/basics/layout.mdx

```diff
@@ -0,0 +1,166 @@
+import * as SplashScreen from 'expo-splash-screen';
+import { useEffect } from 'react';
+
+SplashScreen.preventAutoHideAsync();
+
+export default function RootLayout() {
+  const [loaded] = useFonts({
+    SpaceMono: require('../assets/fonts/SpaceMono-Regular.ttf'),
+  });
+
+  useEffect(() => {
+    if (loaded) {
+      SplashScreen.hideAsync();
+    }
+  }, [loaded]);
+
+  if (!loaded) {
+    return null;
+  }
+
+  return <Stack />;
+}
+```
+
+This example keeps the splash screen showing initially, and then renders a stack navigator once the fonts are loaded, which will cause your app to proceed to the initial route.
+
+## Stacks
+
+You can implement a stack navigator in your root layout, as shown above, or in any other layout file inside of a folder. Let's suppose you have a file structure with a stack inside of a folder:
+
+<FileTree
+  files={[
+    ['app/products/\_layout.tsx'],
+    ['app/products/index.tsx'],
+    ['app/products/[productId].tsx'],
+    ['app/products/accessories/index.tsx'],
+  ]}
+/>
+
+If you want everything inside of the **products** folder to be arranged in a stack relationship, inside the **\_layout.tsx** file, return a `Stack` component:
+
+```tsx _layout.tsx
+import { Stack } from 'expo-router';
+
+export default function StackLayout() {
+  return <Stack />;
+}
+```
+
+When you navigate to `/products`, it will first go to the default route, which is **products/index.tsx**. If you navigate to `/products/123`, then that page will be pushed onto the stack. By default, the stack will render a back button in the header that will pop the current page off the stack, returning the user to the previous page. Even when a page isn't visible, if it is still pushed onto the stack, it is still being rendered.
+
+The `Stack` component implements React Navigation's native stack, and can use the same screen options. However, you do not have to define the pages specifically inside the navigator. The files inside the folder will be automatically treated as eligable routes in the stack. However, if you want to define screen options, you can add a `Stack.Screen` component inside the `Stack` component. The `name` prop should match the route name, but you do not need to supply a `component` prop; Expo Router will map this automatically:
+
+```tsx _layout.tsx
+import { Stack } from 'expo-router';
+
+export default function StackLayout() {
+  return (
+    <Stack>
+      <Stack.Screen name="[productId]" options={{ headerShown: false }} />
+    </Stack>
+  );
+}
+```
+
+While it is possible to nest navigators, be sure to only do so when it is truly needed. In our example, if you wanted to then push **products/accessories/index.tsx** onto the stack, it's not necessary to have an additional **\_layout.tsx** in the **accessories** folder with a `Stack` navigator. That would define another stack inside the first one. It is fine to add folders that only affect the URL but otherwise use the same navigator as the parent folder.
```

Comment:
```suggestion
While it is possible to nest navigators, be sure to only do so when it is truly needed. In the above example, if you want to push **products/accessories/index.tsx** onto the stack, it's not necessary to have an additional **\_layout.tsx** in the **accessories** directory with a `Stack` navigator. That would define another stack inside the first one. It is fine to add directories that only affect the URL, otherwise, use the same navigator as the parent directory.
```